changelog/10465.deprecation.rst

```diff
@@ -0,0 +1 @@
+Test functions returning a value other than None will now return a warning and not an error.
```

Comment:
```suggestion
Test functions returning a value other than None will now issue a `PytestWarning` instead of `PytestRemovedIn8Warning`, meaning this will stay a warning instead of becoming an error in the future.
```